internal/elasticsearch/common.go

```diff
@@ -339,12 +339,16 @@ func newEnvVars(nodeName, clusterName, instanceRAM string, roleMap map[api.Elast
 // TODO: add isChanged check for labels and label selector
 func newLabels(clusterName, nodeName string, roleMap map[api.ElasticsearchNodeRole]bool) map[string]string {
 	return map[string]string{
-		"es-node-client": strconv.FormatBool(roleMap[api.ElasticsearchRoleClient]),
-		"es-node-data":   strconv.FormatBool(roleMap[api.ElasticsearchRoleData]),
-		"es-node-master": strconv.FormatBool(roleMap[api.ElasticsearchRoleMaster]),
-		"cluster-name":   clusterName,
-		"component":      "elasticsearch",
-		"node-name":      nodeName,
+		"es-node-client":               strconv.FormatBool(roleMap[api.ElasticsearchRoleClient]),
+		"es-node-data":                 strconv.FormatBool(roleMap[api.ElasticsearchRoleData]),
+		"es-node-master":               strconv.FormatBool(roleMap[api.ElasticsearchRoleMaster]),
+		"cluster-name":                 clusterName,
+		"component":                    "elasticsearch",
+		"node-name":                    nodeName,
+		"app.kubernetes.io/name":       nodeName,
+		"app.kubernetes.io/component":  "elasticsearch",
```

Comment:
I think these values do not match the [documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels). Reading the documentation I have two interpretations:

| Label | Option 1 | Option 2 |
|-:|:-|:-|
|`app.kubernetes.io/name`|static, for example "elasticsearch"|value of `clusterName`|
|`app.kubernetes.io/component`|dependent on component, for example "elasticsearch", "kibana" and "indexmanagement"|dependent on component, for example "elasticsearch", "kibana" and "indexmanagement"|
|`app.kubernetes.io/instance`|value of `clusterName`|value of `nodeName`|

I don't know if we need the `nodeName` as part of these labels. That the Elasticsearch component is consisting of more than one node can be seen as an "implementation detail" that is not relevant on the level of these labels. This seems to be the closest interpretation of the documentation (Option 1).